avocado/core/job.py

```diff
@@ -539,6 +542,9 @@ def run(self):
                  :mod:`avocado.core.exit_codes` for more information.
         """
         runtime.CURRENT_JOB = self
+        for ext in self.job_pre_post_dispatcher.extensions:
```

Comment:
Actually, that are even more changes to the `Job` class that I want to make.  I have a WiP version that better defines at what point a job has started, along the lines you suggested here.

This is something that I can add to this to this PR, but seemed to me like this is bound to generate even more discussion... related, yes, but necessary for this exact work.  We have not defined (on purpose) what *exactly* means "pre" job... that is even trickier because it would really mean that a job does not exist yet.  Still, we're implicitly (not good, but again, on purpose at this time) that "pre" means "before a job runs tests".